batch.py

```diff
@@ -0,0 +1,389 @@
+
+
+@click.command()
+@click.option("--chromosomes")
+@click.option("--genes")
+@click.option("--celltypes")
+@click.option("--expression-file-prefix")
+@click.option("--sample-mapping-file")
+@click.option("--mt-path")
+@click.option("--anno-ht-path")
+def crm_pipeline(
+    # sc_samples: list["str"],
+    chromosomes: list[str],
+    genes: list[str],
+    celltypes: list[str],
+    expression_files_prefix: str,  # 'scrna-seq/grch38_association_files'
+    sample_mapping_file: str,
+    mt_path: str = DEFAULT_JOINT_CALL_MT,  # 'mt/v7.mt'
+    anno_ht_path: str = DEFAULT_ANNOTATION_HT,  # 'tob_wgs_vep/104/vep104.3_GRCh38.ht'
+    window_size: int = 50000,
+):
+
+    sb = hb.ServiceBackend(
+        billing_project=config["hail"]["billing_project"],
+        remote_tmpdir=remote_tmpdir(),
+    )
+    batch = hb.Batch("CellRegMap pipeline", backend=sb)
+
+    # extract samples for which we have single-cell (sc) data
+    sample_mapping_file = remove_sc_outliers(sample_mapping_file)
+    sc_samples = sample_mapping_file["OneK1K_ID"].unique()
+
+    # filter to QC-passing, rare, biallelic variants
+    filter_job = batch.new_python_job(name="MT filter job")
+    output_mt_path = output_path("tob_wgs_rv/densified_rv_only.mt")
+    filter_job.call(
+        filter_variants,
+        mt_path=mt_path,
+        samples=sc_samples,
+        output_mt_path=output_mt_path,
+    )
+
+    # grab all relevant genes across all chromosomes
+    # simpler if gene details are condensed to one file
+    gene_dict: dict[str, dict] = {}
+    for chromosome in chromosomes:
+        geneloc_tsv_path = os.path.join(
+            expression_files_prefix,
+            "gene_location_files",
+            f"GRCh38_geneloc_chr{chromosome}.tsv",
+        )
+
+        # concatenating across chromosomes to have a single dict
+        gene_dict.update(make_gene_loc_dict(geneloc_tsv_path))
+
+    # isolate to the genes we are interested in
+    genes_of_interest = genes or list[gene_dict.keys()]
+
+    # for each gene, extract relevant variants (in window + with some annotation)
+    # submit a job for each gene (export genotypes to plink)
+    genotype_jobs = []
+    for gene in genes_of_interest:
+
+        # final path for this gene - generate first (check syntax)
+        plink_file = output_path(f"{plink_output_prefix}/{gene}")
+        gene_dict[gene]["plink"] = plink_file
+
+        # if the plink output exists, do not re-generate it
+        if to_path(plink_file).exists():
+            continue
+
+        plink_job = batch.new_python_job(f"Create plink files for: {gene}")
+        plink_job.depends_on(filter_job)
+        plink_job.call(
+            get_promoter_variants,
+            mt_path=output_mt_path,
+            ht_path=anno_ht_path,
+            gene_details=gene_dict[gene],
+            window_size=window_size,
+            plink_file=plink_file,
+        )
+        # concatenate jobs so they can be depended on
+        genotype_jobs.append(plink_job)
```

Comment:
You'll need to do a `batch.run(wait=False)` to set all the jobs running

Reply:
nice yea that makes sense